### ISSUE TITLE
NLog.Web.AspNetCore 4.5.4 - NLog configured in a DLL, class in the DLL only called by other DLL's

### ISSUE THREAD

**Issue by vbossio** [NLog/NLog#2774]
I have a DLL that has NLog configured to write to a sql server database.
Call it DLL-A and the database it writes to DB-A.
It contains a class called CLS-A that is only called from other DLL's (never instantiated by DLL-A).
I have a second DLL that has its own NLog config call it DLL-B and it writes to DB-B.
DLL-B calls CLS-A to do some work.
How do I get CLS-A to log to DB-A, even though it is only called by DLL-B (and still use dependency injection)?

**Comment by snakefoot** [comment]
Maybe a local LogFactory in each DLL with their individual nlog.config:

https://github.com/NLog/NLog/wiki/Configure-component-logging

**Comment by vbossio** [comment]
Thank you for that but the plot thickens...
It's a little more complicated now that I did a deep dive.  Let me rephrase the question:
SVC-0 is a SERVICE that creates an instance of NLog that writes to DB-0.
SVC-0 injects it's logger to DLL-A which writes to DB-0 - no problem so far.
DLL-A contains a class called CLS-A that is only called from other DLL's (never instantiated by DLL-A or SVC-0).
I have another DLL that has its own NLog config call it DLL-B and it writes to DB-B.
DLL-B calls CLS-A to do some work.
How do I get CLS-A to log to DB-0, even though it is only called by DLL-B (and still use dependency injection)?

**Comment by snakefoot** [comment]
Sounds like something you should take to stackoverflow or some other forum.


Sent from my Samsung device


-------- Original message --------

**Comment by 304NotModified** [comment]
I sorry, but I think this is off topic. It's a DI question, so closing this.
